packages/features/ee/workflows/lib/reminders/providers/twilioProvider.ts

```diff
@@ -90,3 +136,44 @@ export const verifyNumber = async (phoneNumber: string, code: string) => {
     }
   }
 };
+
+async function isLockedForSMSSending(userId?: number | null, teamId?: number | null) {
+  if (teamId) {
+    const team = await prisma.team.findFirst({
+      where: {
+        id: teamId,
+      },
+    });
+    return team?.smsLockState === SMSLockState.LOCKED;
+  }
+
+  if (userId) {
+    const memberships = await prisma.membership.findMany({
+      where: {
+        userId: userId,
+      },
+      select: {
+        team: {
+          select: {
+            smsLockState: true,
+          },
+        },
+      },
+    });
+
+    const memberOfLockedTeam = memberships.find(
+      (membership) => membership.team.smsLockState === SMSLockState.LOCKED
+    );
+
+    if (!!memberOfLockedTeam) {
+      return true;
+    }
+
+    const user = await prisma.user.findFirst({
+      where: {
+        id: userId,
+      },
+    });
+    return user?.smsLockState === SMSLockState.LOCKED;
+  }
```

Comment:
NIT: Can we actually use an `OR` condition in the prisma query against the `userId` to see if the user is locked.